### PR TITLE
Add Null VM tests and merge  some tests to decrease compile time.

### PR DESCRIPTION
**Pull request by jplevyak**
This depends on https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/31

Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

**Comment by jplevyak** [comment]
ping

File: test/extensions/bootstrap/wasm/config_test.cc [hunk 1]
```diff
@@ -48,113 +67,69 @@ class WasmFactoryTest : public testing::TestWithParam<std::string> {
 };
 
 INSTANTIATE_TEST_SUITE_P(Runtimes, WasmFactoryTest,
-                         testing::Values("v8"
+                         testing::Values("v8",
 #if defined(ENVOY_WASM_WAVM)
```
Comment (lizan) [inline test/extensions/bootstrap/wasm/config_test.cc:71]: FYI this pattern (`#if` inside macro expansion) breaks MSVC

Comment (PiotrSikora) [reply test/extensions/bootstrap/wasm/config_test.cc:71]: @jplevyak Should we just nuke WAVM support?

Comment (jplevyak) [reply test/extensions/bootstrap/wasm/config_test.cc:71]: Yes.  I am going to do that as soon as I merge this.

**Pull request merged**